src/ulvt/prime_field_sumcheck/utils/interpolate.hpp

```diff
@@ -0,0 +1,9 @@
+#include "../../finite_fields/qm31.cuh"
+
+constexpr QM31 one_half = (uint32_t) 0x40000000;
```

Comment:
Is this constant required outside of `interpolate_at`? If so,
```suggestion
inline constexpr QM31 one_half = (uint32_t) 0x40000000;
```
ensures that multiple definitions of `one_half` are not created. Without this specification, the linker would error due to duplicate symbols for `one_half` if multiple compilation units using this constant were included in the same target.